components/tag-list/tag-list-item-mixin.js

```diff
@@ -131,52 +131,55 @@ export const TagListItemMixin = superclass => class extends LocalizeCoreElement(
 		this.clearable = false;
 		/** @ignore */
 		this.keyboardTooltipItem = false;
-		/** @ignore */
-		this.role = 'listitem';
 		this._displayKeyboardTooltip = false;
 		this._id = getUniqueId();
-		this._tooltipShown = false;
+		this._keyboardTooltipShown = false;
 	}
 
 	firstUpdated(changedProperties) {
 		super.firstUpdated(changedProperties);
 
 		const container = this.shadowRoot.querySelector('.tag-list-item-content');
+
 		this.addEventListener('focus', async(e) => {
 			// ignore focus events coming from inside the tag content
 			if (e.composedPath()[0] !== this) return;
 
-			this._displayKeyboardTooltip = (this.keyboardTooltipItem && !this._tooltipShown);
+			this._displayKeyboardTooltip = (this.keyboardTooltipItem && !this._keyboardTooltipShown);
 			await this.updateComplete;
 
-			/** @ignore */
-			container.dispatchEvent(new FocusEvent('focus', { bubbles: false, cancelable: true }));
+			container.focus();
 		});
+
 		this.addEventListener('blur', () => {
 			this._displayKeyboardTooltip = false;
-			/** @ignore */
-			container.dispatchEvent(new FocusEvent('blur', { bubbles: false, cancelable: true }));
 		});
+
 		this.addEventListener('keydown', this._handleKeydown);
 	}
 
-	handleClearItem(e, clearAll) {
+	_handleClearItem() {
```

Comment:
The only thing that called `handleClearItem` outside this class was `d2l-tag-list`, which passed `true` for `clearAll`, which resulted in nothing actually being done. Further, when dispatched from the item's clear button, `handleFocus` would always be `true`, but it wasn't actually being used. As a result, this method could be made private and cleaned up.